src/Wt/WApplication.C

```diff
@@ -1498,8 +1506,12 @@ WApplication::UpdateLock::UpdateLock(WApplication *app)
   WebSession::Handler *handler = WebSession::Handler::instance();
 
   createdHandler_ = false;
-  if (handler && handler->haveLock() && handler->session() == app->session_)
+  if (!app) {
+    return;
+  }
```

Comment:
Can you remove this bit? We only need the first diff chunk. This chunk is used in JWt, which already does the right thing because this constructor is `private` and the lock can only be obtained with `getUpdateLock()`.

Regards,
Roel

Reply:
Sure thing. Done. 

Regards,
Steven